### ISSUE TITLE
Mismatched Connections

### ISSUE THREAD

**Issue by kpirnie** [sonroyaalmerol/m3u-stream-merger-proxy#119]
```
2024/08/25 11:55:50 Received request from 172.20.0.1:65038 for URL: /stream/ei1hbWF6aW5nLXdvcmxkLW9mLWd1bWJhbGw=
2024/08/25 11:55:50 [DEBUG] Stream attempt 1 out of 5
2024/08/25 11:55:50 Current number of connections for M3U_1: 1
2024/08/25 11:55:51 [DEBUG] Successfully fetched stream from http://myprovider/55698
2024/08/25 11:55:51 [DEBUG] Headers set for response: map[Access-Control-Allow-Origin:[*] Cache-Control:[no-cache] Content-Type:[video/mp2t] Date:[Sun, 25 Aug 2024 15:55:51 GMT] Server:[nginx]]
2024/08/25 11:55:51 Proxying 172.20.0.1:65038 to http://myprovider/55698
2024/08/25 11:55:51 Current number of connections for M3U_1: 2
```

That current number of connections, shows 2.  Why, and is it really using 2 connections to the provider?

> Mismatched connections is a definitely a bug. It should only be using one connection each. It's a sign of a leaking goroutine/worker.

**Comment by aniel300** [comment]
was about to make this lol. here is a reference to a potential cause of this as stated by @sonroyaalmerol https://github.com/sonroyaalmerol/m3u-stream-merger-proxy/issues/116#issuecomment-2308913183

**Comment by sonroyaalmerol** [comment]
I found some points of leaks in the stream handler and fixed it in #118. I'm not sure if that fixes this issue. Should be merged in the dev build by now if you don't mind trying it

**Comment by aniel300** [comment]
trying now

**Comment by aniel300** [comment]
am playing two at the moment but the logs shows 6 `2024/08/25 13:29:31 Current number of connections for M3U_1: 6`

**Comment by sonroyaalmerol** [comment]
Seems like my `ClearConcurrencies` function to reset the counters on restart was not working. I've fixed it on #121. Can you try that one in case it was just storing an old instance of the counter?

**Comment by aniel300** [comment]
seems fixed on my end, thanks.

**Comment by kpirnie** [comment]
Fixed here too.

**Comment by sonroyaalmerol** [comment]
The fix should be released on `0.13.2`. I appreciate the help! Closing this issue now :)
